src/main/java/com/cema/activity/controllers/LocationController.java

```diff
@@ -0,0 +1,201 @@
+
+        if (!authorizationService.isAdmin()) {
+            cuig = authorizationService.getCurrentUserCuig();
+        }
+        log.info("Request to delete location with name {} and cuig {}", locationName, cuig);
+        CemaLocation cemaLocation = locationRepository.findCemaLocationByNameAndEstablishmentCuigIgnoreCase(locationName, cuig);
+        if (cemaLocation != null) {
+            log.info("Location exists, deleting");
+            locationRepository.delete(cemaLocation);
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+        log.info("Location Not found");
+        throw new NotFoundException(String.format("Location %s doesn't exits", locationName));
+    }
+
+    @ApiOperation(value = "Modifies an existent Location")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Location modified successfully"),
+            @ApiResponse(code = 404, message = "The location you were trying to modify doesn't exists")
+    })
+    @PutMapping(value = BASE_URL + "{name}", produces = {MediaType.APPLICATION_JSON_VALUE}, consumes = {MediaType.APPLICATION_JSON_VALUE})
+    public ResponseEntity<Location> updateLocation(
+            @ApiParam(
+                    value = "The name of the location we are looking for.",
+                    example = "Corral 5")
+            @PathVariable("name") String name,
+            @ApiParam(
+                    value = "The cuig of the establishment of the location. If the user is not admin will be ignored.",
+                    example = "321")
+            @RequestParam(value = "cuig") String cuig,
+            @ApiParam(
+                    value = "The location data we are modifying")
+            @RequestBody Location location) {
+
+        if (!authorizationService.isAdmin()) {
+            cuig = authorizationService.getCurrentUserCuig();
+        }
+        log.info("Request to modify location with name {} and cuig {}", name, cuig);
+        CemaLocation cemaLocation = locationRepository.findCemaLocationByNameAndEstablishmentCuigIgnoreCase(name, cuig);
+
+        if (cemaLocation == null) {
+            log.info("Location with name {} and cuig {} doesn't exists", name, cuig);
+            throw new NotFoundException(String.format("Location with name %s doesn't exits", name));
+        }
+
+
+        cemaLocation = locationMapper.updateEntityWithDomain(location, cemaLocation);
+
+        locationRepository.save(cemaLocation);
+
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @ApiOperation(value = "Retrieve all locationes for your cuig", response = Location.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Listed all locationes")
+    })
+    @GetMapping(value = BASE_URL + "list", produces = {MediaType.APPLICATION_JSON_VALUE})
+    public ResponseEntity<List<Location>> listLocations() {
+        String cuig = authorizationService.getCurrentUserCuig();
```

Comment:
I think the CUIG should be sent from the FE

Reply:
it is almost the same, since later I do the listing based on the user role.